auth/bls/bls_test.go

```diff
@@ -0,0 +1,36 @@
+// Copyright (C) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package bls
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/ava-labs/hypersdk/crypto"
+	"github.com/ava-labs/hypersdk/crypto/bls"
+)
+
+func TestBLSSign(t *testing.T) {
```

Comment:
For tests like this it's idiomatic to write a table-driven test ([ref](https://go.dev/wiki/TableDrivenTests)) to make it easy to add/modify tests.

I suspect that we can get a decent table-driven test by using a table like:
```
type test struct {
    name string
    message []byte
    signature []byte
    expected error
}
```

If we do this we should be able to clean up this test - since this test technically covers two different cases (one where the signature is valid, and one where the signature is invalid) and we should be testing for one.

Reply:
Sure, will do! 
Maybe for this purpose we could write fuzz tests, because it seems hard to think about messages that would find edge cases since they should all verify as expected, also the signature cannot really be guessed so we can only cheat here by checking the output and writing it as the test input.

Reply:
But absolutely fine for few cases that checks if verify works/fails as expected ! Thanks!

Reply:
Hmm yeah you're right. In that case you can either have a `setup func() ([]byte, []byte)` that generates the message + signature pair, or we can just split the current UT into two UTs (valid/invalid). We can also do fuzz testing like you mentioned... i think these are all reasonable so pick whatever you think looks best.

Reply:
So I have changes the bls test suite to have an `expectedSignature` in addition to the `signature`.
The `signature` will overwrite the signature that gets written by blst when calling `Sign` so that we can check that the verification fails as expected. Not sure it it's the best solution.